kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -162,12 +162,25 @@ namespace Kratos
 					elemental_distances[i] = approximation_plane.CalculateSignedDistance(r_geometry[i]);
 				}
 			} else {
-				// Create a plane with the 3 intersection points (or 2 in 2D)
-				Plane3D plane = SetIntersectionPlane(int_pts_vector);
-
-				// Compute the distance to the intersection plane
-				for (int i = 0; i < number_of_tetrahedra_points; i++) {
-					elemental_distances[i] = plane.CalculateSignedDistance(r_geometry[i]);
+				if (int_pts_vector.size() >= TDim) {
```

Comment:
Here I check if there is enough intersection points to build a plane.

For instance, in TEST 1 or 4, there is only 1 intersection point, so we can't create a plane with the intersection points.

I use the "IntersectedObject" instead, but Im not sure if this is the best solution